src/GraphQL/Resolver.hs

```diff
@@ -134,17 +133,24 @@ data Result a = Result [ResolverError] a deriving (Show, Functor, Eq)
 aggregateResults :: [Result Value] -> Result Value
 aggregateResults r = toValue <$> sequenceA r
 
+throwE :: Applicative f => ResolverError -> f (Result Value)
+throwE err = pure (Result [err] GValue.ValueNull)
+
 instance Applicative Result where
   pure v = Result [] v
   (Result e1 f) <*> (Result e2 x) = Result (e1 <> e2) (f x)
 
 ok :: Value -> Result Value
 ok = pure
 
+type AllTypes = Map Name TypeDefinition
 
 class HasResolver m a where
   type Handler m a
-  resolve :: Handler m a -> SelectionSet Value -> m (Result Value)
+  -- XXX: Might be nicer to make AllTypes a ReaderT -- easier to ignore when
+  -- not wanted, and wouldn't have required me to change so many call sites.
+  -- OTOH, with all our type weirdness, I got bored trying to figure it out.
+  resolve :: AllTypes -> Handler m a -> Maybe (SelectionSetByType Value) -> m (Result Value)
```

Comment:
Cool. I'll make `AllTypes` a `newtype` with only a lookup operation so as to make this a bit more obvious.